README-dev.md: add tips when updating the bindings (#1029)
o1-labs/o1js: README-dev.md

## Run tests

> If you are updating the bindings, you might need to clean first your
> environment using `npm run clean`.

Reviewer: ```suggestion
If you are have recently changed branches, you might need to clean first your
environment using `npm run clean`.
```
how about this? it's more general and doesn't have to do with updating the bindings. Actually, `npm run bindings` does run `npm run clean`

Author: Your suggestion makes more sense as it is what certainly happened to me.
